aptos-move/aptos-vm/src/move_vm_ext/respawned_session.rs

```diff
@@ -254,7 +254,8 @@ impl<'r> TResourceGroupView for ExecutorViewWithChangeSet<'r> {
         resource_tag: &Self::ResourceTag,
         maybe_layout: Option<&Self::Layout>,
     ) -> anyhow::Result<Option<Bytes>> {
-        if let Some(write_op) = self
+        // TODO: resource_group_write_set also contains a layout. What to do with it?
+        if let Some((write_op, _layout)) = self
```

Comment:
seems like on line 233 we also ignore the layout. 

I assume at this point, it is required to be the same layout. maybe we should just have a function expect_layouts_match(maybe_layout_1, maybe_layout_2)

which we call here and there, and then inside we add a TODO of what exactly to do, but we can start with asserting that both are either None, or both are Some. 

we can later also do something like - randomly for 1% of calls, we check fully. when we run tests, we can locally change to always compare, etc. but there will be only a single place to change it. you can put that function in aptos-move/aptos-aggregator/src/types.rs next to expect_ok, or if you see a better place elsewhere

Reply:
Why should the layout in resource_group_write_set, and the layout used by read operations be the same? If the current transaction has updated the layout, then the new layout will be present in the resource_group_write_set. Will the read operation use the new layout or the old layout?